designs/cloudWatch/cloudWatchIntegration.md

```diff
@@ -0,0 +1,76 @@
+# CloudWatch Logs Experience
+
+* **Type**: UX Design
+* **Status**: Proposal
+
+## Introduction
+CloudWatch Logs enables a user to centralize the logs from all of their systems, applications, and AWS services that they use, in a single, highly scalable service. This document describes adding CloudWatch Logs into the JetBrains toolkit in order to facilitate easily viewing and searching them for specific error codes or patterns.
+
+## Comparison to other products
+
+| Product                       | Difference    |
+|-------------------------------|---------------|
+| AWS Toolkit for Visual Studio | Not supported |
+| AWS Toolkit for VSCode        | Not supported |
+
+## Experience
+
+### View Log Groups
+A new AWS Explorer root node would be added for the CloudWatch service. Underneath of that node we would add the list of Log Groups that exist in the selected region for the selected credential profile. Double clicking a Log Group would open the CloudWatch tool window with the tab for the Log Streams that belong to that group. A new tab would be created if needed.
+
+*Browsing Log Groups in AWS Explorer:*<br />
+![BROWSE_LOG_GROUP]
+
+#### Alternative Proposal
+Instead of building a new CloudWatch service node in the AWS Explorer, we instead build a CloudWatch Log Group browser tab in the CloudWatch tool window. This new tab would also be linked to the active AWS connection settings. This would differ from the other tabs which may lead to a confusing user experience and seen as a con.
+
+*Possible alternative for browsing Log Groups:*<br />
+![ALT_BROWSE_LOG_GROUP]
+
+### View Log Streams
+Upon selecting a Log Group, the user would be shown a new tab in the CloudWatch tool window. This new tab will list all the streams for the requested group to allow the user to find the exact stream they are looking for.
+
+Log Streams can also be sorted by name or last event time in order to help facilitate finding the Log Stream of interest.
+
+*View all Log Streams in the selected Log Group:*<br />
+![BROWSE_LOG_STREAMS]
+
+#### Navigate to a Log Stream
+Double clicking the Log group in the AWS Explorer will open the Log Stream selection. 
+
+For additional service integrations, such as a Lambda function, where we know what Log Group is associated with the resource a new context menu action adds the ability to bypass having to search for the Log Group and instead open the Log Stream selection directly.
```

Comment:
I think this is going to be the more common usage pattern when working with systems that have this integration (like Lambda functions).

Run on sentence? How about...

Some services will have a more intuitive and direct association with CloudWatch Logs. For example, a Lambda Function context menu can take users directly to the function's log streams.

Reply:
A supplemental experience could be to add a sub-node to the Lambda function. Logs could be considered something that the Lambda "has".

Reply:
> A supplemental experience could be to add a sub-node to the Lambda function. Logs could be considered something that the Lambda "has".

We have no active prior art on this, we used to have Lambdas that belong to a CloudFromation stack underneath of it but we removed that for technical reasons. Your idea interests me and would probably be a good question to ask in a future UX study

Reply:
Definitely an interesting concept.